agent/structs/config_entry_status.go

```diff
@@ -195,3 +209,308 @@ func (u *StatusUpdater) UpdateEntry() (ControlledConfigEntry, bool) {
 	u.entry.SetStatus(u.status)
 	return u.entry, true
 }
+
+// GatewayConditionType is a type of condition associated with a
+// Gateway. This type should be used with the GatewayStatus.Conditions
+// field.
+type GatewayConditionType string
```

Comment:
so I'm also thinking that either these are in the wrong package (should be in `agent/api/config_entry_status`) or we also need to export these from the API package so we can use them from other places programatically